packages/filesystem/src/node/nsfw-watcher/nsfw-filesystem-watcher.ts

```diff
@@ -111,17 +111,17 @@ export class NsfwFileSystemWatcherServer implements FileSystemWatcherServer {
         let watcher: nsfw.NSFW | undefined = await nsfw(fs.realpathSync(basePath), (events: nsfw.ChangeEvent[]) => {
             for (const event of events) {
                 if (event.action === nsfw.actions.CREATED) {
-                    this.pushAdded(watcherId, paths.join(event.directory, event.file!));
+                    this.pushAdded(watcherId, this.resolvePath(event.directory, event.file!));
                 }
                 if (event.action === nsfw.actions.DELETED) {
-                    this.pushDeleted(watcherId, paths.join(event.directory, event.file!));
+                    this.pushDeleted(watcherId, this.resolvePath(event.directory, event.file!));
                 }
                 if (event.action === nsfw.actions.MODIFIED) {
-                    this.pushUpdated(watcherId, paths.join(event.directory, event.file!));
+                    this.pushUpdated(watcherId, this.resolvePath(event.directory, event.file!));
```

Comment:
Q: shouldn't both be reported as changed? I mean the link and the target.

Reply:
i would need to check whether it is possible to report for all symlins, if not then we stick only with the real part. reportin fs events should be fast, detecting all symlinks would harm perf

Reply:
I checked with vscode behavior, and this change would make it quite similar.